homeassistant/components/triggercmd/config_flow.py

```diff
@@ -0,0 +1,72 @@
+"""Config flow for TRIGGERcmd integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import jwt
+from triggercmd import client
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, TOKEN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({(TOKEN): str})
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> str:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    if len(data[TOKEN]) < 100:
+        raise InvalidToken
+
+    tokenData = jwt.decode(data[TOKEN], options={"verify_signature": False})
+    if not tokenData["id"]:
+        raise InvalidToken
+
+    r = await client.async_list(data[TOKEN])
+    if not r.status_code == 200:
+        raise InvalidToken
+
+    return tokenData["id"]
+
+
+class TriggerCMDConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                title = await validate_input(self.hass, user_input)
```

Comment:
We should also try to do a connection check during setup, just to make sure we are able to setup. This way we only create an entry when we know the integration for sure is going to work

Reply:
I added the connection check with these lines (77-80):
```
                errors = await test_connection(
                    errors=errors, token=user_input[CONF_TOKEN]
                )
                if errors.get("connection") != "connection_error":
```